docs/proposal/2022-11-22-botkube-plugins-integration.md

```diff
@@ -0,0 +1,498 @@
+       huseyinbabal: https://raw.githubusercontent.com/huseyinbabal/botkube-plugins/main/index.json
+   ```
+
+2. The `executors` definition now can refer to the executor plugins specified in a given repository.
+   ```yaml
+   executors:
+     'plugin-based':
+       botkube/kubectl:     # <repo>/<plugin> is syntax for plugin based executors
+         enabled: true      # if not enabled we don't download and start a given plugin
+         version: v1.0.0    # if empty, the latest version is used
+         config:            # plugin specific configuration
+           namespaces:
+             include: ["botkube", "default","ambassador"]
+           commands:
+             verbs: ["get","logs","top"]
+             resources: ["pods","deployments","nodes","configmap"]
+   ```
+
+3. The `sources` definition now can refer to the source plugins specified in a given repository.
+   ```yaml
+   sources:
+     'plugin-based':
+       botkube/kubernetes:
+         enabled: true
+         version: v1.0.0
+         config:
+           recommendations:
+             pod:
+               noLatestImageTag: true
+               labelsSet: true
+   ```
+
+## Use cases
+
+This section describes example configurations that enable the requested use-cases.
+
+### Defining list of plugin repositories and enabling a given plugin
+
+We introduce a new `plugins` object to Botkube configuration syntax. It holds the list of the plugins repositories. Each repository has own name to prevent conflict with the same plugins name, e.g. both provides `kubectl` plugin.
+
+To enable a given plugin you need to specify it under the `executors` or `sources` property. In the first phase, we only allow to enable and disable plugins using Botkube configuration. Later we can introduce `@Botkube enable [executor|source] NAME` and `@Botkube disable [executor|source] NAME`. Disabling and enabling plugins should be possible in the Botkube runtime without restarting it.
+
+**Syntax:**
+
+```yaml
+executors:
+  'plugin-based':
+    botkube/kubectl:     # <repo>/<plugin> is syntax for plugin based executors
+      enabled: true      # if not enabled we don't download and start a given plugin
+      version: v1.0.0    # if empty, the latest version is used.
+      config:            # plugin specific configuration
+      # ...
+sources:
+  'plugin-based':
+    botkube/kubernetes:
+      enabled: true
+      version: v1.0.0
+      config:
+      # ...
+```
+
+In this way you can:
+
+1. Register multiple plugins repositories.
+2. Enable different plugin version from the same repository.
+3. Enable the same plugin (e.g. `kubectl`) from different repositories.
+
+### Defining executors aliases
+
+Currently, the aliases are built-in into Botkube core logic. One option is to define them in the index file for a given executor:
+
+```yaml
+entries:
+  - name: "kubectl"
+    type: "executor"
+    description: "Kubectl executor plugin."
+    version: "v1.0.0"
+    aliases: [ "kc", "k" ]
```

Comment:
Please see my previous comment (posted for line 182). Shouldn't we also list here `kubectl`? I think it would be better to list all the possibilities here and I don't see any obstacles to not treat all the 3 options the same way 🤔 WDYT?